File: components/credentials/src/main/java/org/cloudfoundry/credhub/requests/CertificateSetRequest.java
package org.cloudfoundry.credhub.requests;


import java.util.Objects;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;

import com.fasterxml.jackson.annotation.JsonProperty;
import org.cloudfoundry.credhub.ErrorMessages;
import org.cloudfoundry.credhub.credential.CertificateCredentialValue;

@SuppressWarnings("unused")
public class CertificateSetRequest extends BaseCredentialSetRequest<CertificateCredentialValue> {

  @NotNull(message = ErrorMessages.MISSING_VALUE)
  @Valid
  @JsonProperty("value")
  private CertificateCredentialValue certificateValue;

  public CertificateCredentialValue getCertificateValue() {
    return certificateValue;
  }

  public void setCertificateValue(
    final CertificateCredentialValue certificateValue) {
    this.certificateValue = certificateValue;
  }

  @Override
  public CertificateCredentialValue getCredentialValue() {
    return certificateValue;
  }

  @Override
  public GenerationParameters getGenerationParameters() {
    return null;
  }

  @Override
  public boolean equals(final Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    final CertificateSetRequest that = (CertificateSetRequest) o;
    return Objects.equals(certificateValue, that.certificateValue);
  }

  @Override
  public int hashCode() {
    return Objects.hash(certificateValue);
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/requests/CertificateSetRequest.kt
package org.cloudfoundry.credhub.requests

import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.validation.Valid
import jakarta.validation.constraints.NotNull
import org.cloudfoundry.credhub.ErrorMessages
import org.cloudfoundry.credhub.credential.CertificateCredentialValue
import java.util.Objects

class CertificateSetRequest : BaseCredentialSetRequest<CertificateCredentialValue>() {
    @NotNull(message = ErrorMessages.MISSING_VALUE)
    @Valid
    @JsonProperty("value")
    var certificateValue: CertificateCredentialValue? = null

    override val generationParameters: GenerationParameters?
        get() = null

    override val credentialValue: CertificateCredentialValue
        get() {
            return this.certificateValue!!
        }

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || javaClass != other.javaClass) {
            return false
        }
        val that = other as CertificateSetRequest?
        return certificateValue == that!!.certificateValue
    }

    override fun hashCode(): Int = Objects.hash(certificateValue)
}
